pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -753,6 +758,25 @@ type Artifact struct {
 type PodGC struct {
 	// Strategy is the strategy to use. One of "OnPodCompletion", "OnPodSuccess", "OnWorkflowCompletion", "OnWorkflowSuccess"
 	Strategy PodGCStrategy `json:"strategy,omitempty" protobuf:"bytes,1,opt,name=strategy,casttype=PodGCStrategy"`
+	// LabelSelector is the label selector to check if the pods match the labels before being added to the pod GC queue.
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,2,opt,name=labelSelector"`
+}
+
+// Matches returns whether the pod labels match with the label selector specified in podGC.
+func (podGC *PodGC) Matches(labels labels.Set) (bool, error) {
+	selector, err := metav1.LabelSelectorAsSelector(podGC.LabelSelector)
```

Comment:
should this return true or false if PodGC == nil or if PodGC.LabelSelector == true?

Reply:
Yes and this would work well with the PodPhase change too.